src/main/java/com/google/api/generator/gapic/protoparser/HttpRuleParser.java

```diff
@@ -200,4 +212,18 @@ private static void checkHttpFieldIsValid(String binding, Message inputMessage,
         fieldCondition,
         String.format(messageFormat, field.name(), inputMessage.name(), field.type()));
   }
+
+  private static Map<String, String> constructPathValuePatterns(String pattern) {
+    Map<String, String> varPattern = new HashMap<>();
```

Comment:
The same question.

Reply:
Order does not matter (see other response for details)